### ISSUE TITLE
Enumerate difficulties with current API

### ISSUE THREAD

**Issue by pittst3r** [bigtestjs/bigtest#13]
Part of quest issue https://github.com/bigtestjs/server/issues/35

Solutions we come up with should have resolutions for these issues with the 1.0 API.

- Action properties, getter properties, etc are confusing.
- No way to cancel a running interactor. For example, if the test fails for another reason, either a timeout or a failed assertion running in parallel, there's no way to bail early without running the full course of assertions.
- Difficult to type strongly/high res.
- Decorators are required for some features, but they are [not even close to fully baked](https://github.com/tc39/proposal-decorators#faq).

**Comment by pittst3r** [comment]
@JohnC-80 Not sure I follow. Is the problem that you are needing to add wrappers to components just for testing?

I looked at the codesandbox fork you linked to and the only API change I saw was the addition of these two properties but I'm not sure what they're for exactly:

```ts
export declare class Interactor {
  // ...
  static _container: string;
  static _defaultScope: string;
}
```

**Comment by JohnC-80** [comment]
@pittst3r Yes that's one part of it. If possible, it would be nice to reduce the impact that interactors/testing have on the application's code - On our project we see alot of extra `<div data-test-attribute>` added in only for sake of testing. Additionally, as someone who writes interactors at the component level, it would be nice to be able to define the outer scope that selectors within an interactor can function within. Say I have a special type of component that renders an input, label etc... it would be nice to be able to use the interactor via the id attribute that's ultimately assigned to the input (it's not the outer-most element) and still have the API for the interactor work as it needs to.

My thinking with ` _container` is that it could function as the established outer scope that is implemented within the component (not at the app level) so that a provided initial selector doesn't necessarily have to be for the outer-most element.
This is different from default scope functionality as the `_container` isn't replaced by the initial selector.

I did include an implementation of how it would probably be used in the `test.ts` of the codesandbox as well as an example of the kind of markup/component this would be applicable for in the `index.html` file just to illustrate the use-case.

`_defaultScope` is legacy interactor. It's really just there to differentiate.
